src/hooks/useLoading.js

```diff
@@ -0,0 +1,16 @@
+import { useState } from "react";
+
+function useLoading() {
+
```

Comment:
maybe you could bring in your loading animation/component into your loading hook? (disclaimer because sometimes people think i know the answer when i ask leading questions: this is not one of those times, i do not know, but wouldn't it be nice if we could! maybe you can)